Cura.proto

```diff
@@ -61,6 +61,23 @@ message Polygon {
     float line_width = 3; // The width of the line being laid down
 }
 
+message LayerOptimized {
+    int32 id = 1;
+    float height = 2; // Z position
+    float thickness = 3; // height of a single layer
+
+    repeated PathSegment path_segment = 4; // layer data
```

Comment:
Why are the paths in a layer split into `PathSegment`s?

You could suffice with just the byte arrays below, right?

For what kind of optimization do you use this?